tools/generate_pv_yaml_rest.sh

```diff
@@ -0,0 +1,328 @@
+  -F | --fileset)
+    FSETNAME="$2"
+    shift 2
+    ;;
+  -s | --size)
+    VOLSIZE="$2"
+    shift 2
+    ;;
+  -p | --pvname)
+    VOLNAME="$2"
+    shift 2
+    ;;
+  -c | --storageclass)
+    CLASS="$2"
+    shift 2
+    ;;
+  -a | --accessmode)
+    ACCESSMODE="$2"
+    shift 2
+    ;;
+  -u | --username)
+    USERNAME="$2"
+    shift 2
+    ;;
+  -t | --password)
+    PASSWORD="$2"
+    shift 2
+    ;;
+  -r | --route)
+    URL="$2"
+    shift 2
+    ;;
+  --)
+    shift
+    break
+    ;;
+  *)
+    usage
+    exit 1
+    ;;
+  esac
+done
+
+# Check for mandatory Params
+MPARAM=""
+[[ -z "${FSNAME}" ]] && MPARAM="${MPARAM}--filesystem "
+[[ -z "${VOLSIZE}" ]] && MPARAM="${MPARAM}--size "
+
+if [ ! -z "$MPARAM" ]; then
+  echo "ERROR: Mandatory parameter missing : $MPARAM"
+  usage
+fi
+
+if [[ ! -z "${VOLPATH}" && ! -z "${FSETNAME}" ]]; then
+  echo "ERROR: Missing parameter. Either 'linkpath' or 'fileset' is mandatory"
+  usage
+fi
+
+if [[ ! ${VOLSIZE} =~ ^[1-9][0-9]*$ ]]; then
+  echo "ERROR: Provided value for --size=${VOLSIZE} is not valid number"
+  exit 2
+fi
+
+if [[ ${#VOLNAME} -ge 254 ]]; then
+  echo "ERROR: pvname specified against option --pvname must be less than 254 characters"
+  exit 2
+fi
+
+if [ -z "${VOLNAME}" ]; then
+  VOLNAME=${VOLPATH%/}
+  VOLNAME=${VOLNAME##*/}
+  VOLNAME="pv-${FSNAME}-${VOLNAME}"
+  VOLNAME=${VOLNAME,,}
+  if [[ ${#VOLNAME} -ge 254 ]]; then
+    echo "ERROR: Specify name for pv using option --pvname"
+    exit 2
+  fi
+
+  if ! [[ "${VOLNAME}" =~ ^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$ ]]; then
+    echo "ERROR: Specify name for pv using option --pvname"
+    exit 2
+  fi
+fi
+
+if ! [[ "${VOLNAME}" =~ ^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$ ]]; then
+  echo "ERROR: Invalid pv name specified. pv name must satisfy DNS-1123 label requirement."
+  exit 2
+fi
+
+[[ -z "${ACCESSMODE}" ]] && ACCESSMODE="ReadWriteMany"
+
+if ! [[ "$ACCESSMODE" == "ReadWriteMany" || "$ACCESSMODE" == "ReadWriteOnce" ]]; then
+  echo "ERROR: Invalid access mode specified. Valid accessmode are ReadWriteMany and ReadWriteOnce."
+  exit 2
+fi
+
+STORAGECLASS=""
+if ! [[ -z "${CLASS}" ]]; then
+  if ! [[ "${CLASS}" =~ ^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$ ]]; then
+    echo "ERROR: Invalid storageClass name specified. storageClass name must satisfy DNS-1123 label requirement."
+    exit 2
+  fi
+  STORAGECLASS="storageClassName: ${CLASS}"
+fi
+
+# Check if this is spectrum scale node
+#if [[ ! -f /usr/lpp/mmfs/bin/mmlscluster ]]; then
+#  echo "ERROR: Spectrum Scale cli's are not present on this node"
+#  exit 2
+#fi
+
+echo >${ERROROUT}
+
+# Get the Spectrum Scale cluster ID
+clusterID=$(curl -k -u "${USERNAME}":"${PASSWORD}" -X GET \
+  --header 'accept:application/json' \
+  "https://${URL}:443/scalemgmt/v2/cluster" \
+  2>${ERROROUT} | python3 -c "import sys, json; print(json.load(sys.stdin)['cluster']['clusterSummary']['clusterId'])")
```

Comment:
should we check python3 dependency at first step ?

Reply:
Pre-requisite check is added.